rav1e_js/www/src/App.tsx

```diff
@@ -0,0 +1,118 @@
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+import React, { useState } from 'react';
+import { Encoder, EncoderConfig, Packet } from "rav1e";
+
+// Configuration for encoder
+const conf = new EncoderConfig()
+	.setDim(64, 96)
+	.setRdoLookaheadFrame(1);
+// JSON-string representing configuration
+const config_str = JSON.stringify(JSON.parse(conf.toJSON()), null, 4);
+
+// Encoder
+const enc = new Encoder(conf);
+
+// Frame to send to encoder
+const f = enc.newFrame();
+
+export default function App() {
+	const [encoderStatus, setEncoderStatus] = useState("");
+	const [flushing, setFlushing] = useState(false);
+	const [frameQ_length, setFrameQ_length] = useState(0);
+	const [packets, setPackets] = useState(Array<Packet>())
+
+	const sendFrame = () => {
+		enc.sendFrame(f);
+		setFrameQ_length(frameQ_length + 1);
+	};
+
+	const flush = () => {
+		enc.flush();
+		setFlushing(true);
+	}
+
+	const receivePacket = () => {
+		try {
+			const p = enc.receivePacket();
+			console.log(p.display());
+			setEncoderStatus("");
+
+			// append new packet
+			const newPackets = Array.from(packets);
+			newPackets.push(p);
+			setPackets(newPackets);
+		} catch (e) {
+			if (e === "encoded") {
+				receivePacket();
+			} else if (e === "need more data") {
+				setEncoderStatus(e + " (send more frames or flush the encoder)");
+			}
+		}
+	}
+
+	return (
+		<div>
+			<h1>Rav1e_js demo</h1>
+
+			<button disabled={flushing} onClick={sendFrame}>Send Frame</button>
+			<button disabled={flushing || frameQ_length === 0} onClick={flush}>Flush</button>
+			<button disabled={(packets.length === frameQ_length)} onClick={receivePacket}>Receive Packet</button>
+
+			<p>Frames in queue: {frameQ_length - packets.length}</p>
+			<p>Flushing: {String(flushing)}</p>
+			<p>Encoder Status: {encoderStatus}</p>
+
+			<details>
+				<summary>About and Help</summary>
+
+				<h3>About</h3>
+				<p>Thank you for trying the <a href="https://github.com/xiph/rav1e">rav1e</a>-encode-in-browser demo ^.^</p>
+				<p>You are able to send frames to the encoder, encode them and view a list of the emitted Packets. And the best thing? Everything is running locally in your browser thanks to <a href="https://webassembly.org">webassembly</a>!</p>
+
+				<h3>Buttons</h3>
+				<table>
+					<tbody>
+						<tr>
+							<td><strong>Name</strong></td>
+							<td><strong>Functionality</strong></td>
+							<td><strong>Details</strong></td>
+						</tr>
+						<tr>
+							<td>Send Frame</td>
+							<td>Send a new (blank) frame to the encoder.</td>
+							<td></td>
+						</tr>
+						<tr>
+							<td>Flush</td>
+							<td>Close the encoder.</td>
```

Comment:
```suggestion
							<td>Siginals the End of the Video.</td>
```
As per encoder.rs 
` Flushing signals the end of the video. After the encoder has been flushed, no additional frames are accepted.`
I am not quite sure if Close the encoder is the right word to use here.